plugins/inputs/logstash/logstash_test.go

```diff
@@ -0,0 +1,364 @@
+	}
+	fakeServer.Listener, _ = net.Listen("tcp", fmt.Sprintf("%s:%s", requestURL.Hostname(), requestURL.Port()))
+	fakeServer.Start()
+	defer fakeServer.Close()
+
+	if logstashTest.client == nil {
+		client, err := logstashTest.createHTTPClient()
+
+		if err != nil {
+			t.Logf("Can't createHttpClient")
+		}
+		logstashTest.client = client
+	}
+
+	if err := logstashTest.gatherPipelineStats(logstashTest.LogstashURL+pipelineStats, &accPipelineStats); err != nil {
+		t.Logf("Can't gather Pipeline stats")
+	}
+
+	if !accPipelineStats.HasMeasurement("logstash_events") {
+		t.Errorf("acc.HasMeasurement: expected logstash_events")
+	}
+
+	if !accPipelineStats.HasMeasurement("logstash_plugin_output_stdout") {
+		t.Errorf("acc.HasMeasurement: expected logstash_plugin_output_stdout")
+	}
+
+	return
+}
+
+func Test_gatherProcessStats(t *testing.T) {
+	fakeServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", string(processJSON))
+	}))
+	requestURL, err := url.Parse(logstashTest.LogstashURL)
+	if err != nil {
+		t.Logf("Can't connect to: %s", logstashTest.LogstashURL)
+	}
+	fakeServer.Listener, _ = net.Listen("tcp", fmt.Sprintf("%s:%s", requestURL.Hostname(), requestURL.Port()))
+	fakeServer.Start()
+	defer fakeServer.Close()
+
+	if logstashTest.client == nil {
+		client, err := logstashTest.createHTTPClient()
+
+		if err != nil {
+			t.Logf("Can't createHttpClient")
+		}
+		logstashTest.client = client
+	}
+
+	if err := logstashTest.gatherProcessStats(logstashTest.LogstashURL+processStats, &accProcessStats); err != nil {
+		t.Logf("Can't gather JVM stats")
+	}
+
+	if !accProcessStats.HasMeasurement("logstash_process") {
+		t.Errorf("acc.HasMeasurement: expected logstash_process")
+	}
+
+	return
+}
+
+func Test_gatherJVMStats(t *testing.T) {
+	fakeServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", string(jvmJSON))
+	}))
+	requestURL, err := url.Parse(logstashTest.LogstashURL)
+	if err != nil {
+		t.Logf("Can't connect to: %s", logstashTest.LogstashURL)
+	}
+	fakeServer.Listener, _ = net.Listen("tcp", fmt.Sprintf("%s:%s", requestURL.Hostname(), requestURL.Port()))
+	fakeServer.Start()
+	defer fakeServer.Close()
+
+	if logstashTest.client == nil {
+		client, err := logstashTest.createHTTPClient()
+
+		if err != nil {
+			t.Logf("Can't createHttpClient")
+		}
+		logstashTest.client = client
+	}
+
+	if err := logstashTest.gatherJVMStats(logstashTest.LogstashURL+jvmStats, &accJVMStats); err != nil {
+		t.Logf("Can't gather JVM stats")
+	}
+
+	if !accJVMStats.HasMeasurement("logstash_jvm") {
+		t.Errorf("acc.HasMeasurement: expected logstash_jvm")
+	}
+
+	return
+}
+
+func Test_Gather(t *testing.T) {
+
+	Test_gatherJVMStats(t)
+	Test_gatherPipelineStats(t)
+	Test_gatherProcessStats(t)
```

Comment:
I don't understand this test layout, why are we running the tests again?

Maybe it would make more sense with the current testdata to have a single test?  Feel free to use [testify](https://github.com/stretchr/testify) to help reduce the test verbosity.

I think a good example that is similar is the [nginx plus tests](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/nginx_plus/nginx_plus_test.go).